js_modules/dagster-ui/packages/ui-core/src/instance/flattenCodeLocationRows.tsx

```diff
@@ -1,25 +1,56 @@
-import {CodeLocationRowType} from '../workspace/VirtualizedCodeLocationRow';
+import {
+  CodeLocationRowStatusType,
+  CodeLocationRowType,
+} from '../workspace/VirtualizedCodeLocationRow';
 import {WorkspaceLocationNodeFragment} from '../workspace/types/WorkspaceQueries.types';
 
 const flatten = (locationEntries: WorkspaceLocationNodeFragment[]) => {
   // Consider each loaded repository to be a "code location".
   const all: CodeLocationRowType[] = [];
   for (const locationNode of locationEntries) {
     const {locationOrLoadError} = locationNode;
+    let status: CodeLocationRowStatusType;
+    if (locationNode.loadStatus === 'LOADING') {
+      if (locationOrLoadError) {
+        status = 'Updating';
+      } else {
+        status = 'Loading';
```

Comment:
whats the distinction between these two states exactly?

Reply:
Loading is the first time we see it. Just copying the logic from here: https://github.com/dagster-io/dagster/blob/a41c42e2981fd86c0fae66a73823db3acc942169/js_modules/dagster-ui/packages/ui-core/src/workspace/CodeLocationRowSet.tsx#L119

Reply:
lost to phabricator history https://github.com/dagster-io/dagster/commit/c2056577761ca735eebea31eba030658e21459e4